docs/data-validation.md

```diff
@@ -0,0 +1,218 @@
+| ---- | --------------- | ----------- |
+| `"numeric"` | [`NumericValidator`](https://surveyjs.io/Documentation/Library?id=NumericValidator) | Throws an error if the answer is not a number or if an entered number is outside the [`minValue`](https://surveyjs.io/Documentation/Library?id=NumericValidator#minValue) and [`maxValue`](https://surveyjs.io/Documentation/Library?id=NumericValidator#maxValue) range. |
+| `"text"` | [`TextValidator`](https://surveyjs.io/Documentation/Library?id=TextValidator) | Throws an error if the length of entered text is outside the range between [`minLength`](https://surveyjs.io/Documentation/Library?id=TextValidator#minLength) and [`maxLength`](https://surveyjs.io/Documentation/Library?id=TextValidator#maxLength). |
+| `"email"` | [`EmailValidator`](https://surveyjs.io/Documentation/Library?id=EmailValidator) | Throws an error if an entered value is not a valid e-mail address. |
+| `"expression"` | [`ExpressionValidator`](https://surveyjs.io/Documentation/Library?id=ExpressionValidator) | Throws an error when the [`expression`](https://surveyjs.io/Documentation/Library?id=ExpressionValidator#expression) evaluates to `false` (see [Expressions](/Documentation/Library?id=design-survey-conditional-logic#expressions)). |
+| `"answercount"` | [`AnswerCountValidator`](https://surveyjs.io/Documentation/Library?id=AnswerCountValidator) | Throws an error if a user selects fewer choices than specified by [`minCount`](https://surveyjs.io/Documentation/Library?id=AnswerCountValidator#minCount) or more choices than specified by [`maxCount`](https://surveyjs.io/Documentation/Library?id=AnswerCountValidator#maxCount). Applies only to question types that can have multiple values (for instance, [Checkbox](https://surveyjs.io/Documentation/Library?id=questioncheckboxmodel)). |
+| `"regex"` | [`RegexValidator`](https://surveyjs.io/Documentation/Library?id=RegexValidator) | Throws an error if an entered value does not match a regular expression defined in the [`regex`](https://surveyjs.io/Documentation/Library?id=RegexValidator#regex) property. |
+
+- [View the "Built-In Validators" example](https://surveyjs.io/Examples/Library?id=validators-standard)
+- [View the "Expression Validator" example](https://surveyjs.io/Examples/Library?id=validators-expression) 
+
+## Implement Custom Client-Side Validation
+
+SurveyJS Library raises the [`onValidateQuestion`](https://surveyjs.io/Documentation/Library?id=surveymodel#onValidateQuestion) event that you can handle to add custom validation logic to your survey. For example, the following code checks that the answer to a `"memo"` question contains the word "survey":
+
+```js
+import { Model } from "survey-core";
+const surveyJson = {
+  "elements": [{
+    "name": "memo",
+    "type": "comment",
+    // ...
+  }]
+};
+const survey = new Model(surveyJson);
+
+survey.onValidateQuestion.add((survey, options) => {
+  if (options.name === "memo") {
+    if (!!options.value.indexOf("survey")) {
```

Comment:
May be it should be following condition
if (options.value.indexOf("survey") == -1) {